spec/esv_spec.rb

```diff
@@ -72,6 +72,57 @@
       expect(output[0][1].class).to eq String
     end
 
+    context "when given a header_converters: option" do
+      let(:data) {
+        ESV.generate do |esv|
+          esv << [ "Dogs", "Cats cats" ]
+          esv << [ 1, 2 ]
+        end
+      }
+
+      it "lets you specify header_converters by symbolic name" do
+        output = ESV.parse(data, header_converters: :downcase)
+
+        expect(output).to eq [
+          [ "dogs", "cats cats" ],
+          [ 1, 2 ],
+        ]
+      end
+
+      it "lets you specify header_converters as a proc" do
+        output = ESV.parse(data, header_converters: ->(value) { value.upcase })
+
+        expect(output).to eq [
+          [ "DOGS", "CATS CATS" ],
+          [ 1, 2 ],
+        ]
+      end
+
+      it "lets you register new symbolic names for header_converters and use them" do
+        ESV::HEADER_CONVERTERS[:upcase] = ->(value) { value.upcase }
+
+        output = ESV.parse(data, header_converters: :upcase)
+
+        expect(output).to eq [
+          [ "DOGS", "CATS CATS" ],
+          [ 1, 2 ],
+        ]
+
+        ESV::HEADER_CONVERTERS.delete(:upcase)
```

Comment:
Minor: I wonder if putting this in an `ensure` would work to ensure it runs even if that spec fails. Or would we need to use `after`?

Reply:
I'd say it's easier to use ans read an after block for this.

@olleolleolle Add more contexts!